src/utils/date.ts

```diff
@@ -10,4 +11,127 @@ export function isDate(dateString?: ValueType) {
   }
 
   return !isNaN(new Date(dateString).getDate());
-}
\ No newline at end of file
+}
+
+export enum DATE_FORMATS {
+  TIME = "h:mm A",
+  SHORT = "MMM DD",
+  LONG = "h:mm A, MMM DD, YYYY",
+  TIME_RANGE = "YYYY-MM-DD HH:mm"
+}
+
+type formatDateOptions = {
+  suffix?: string;
+  asTimestamp?: boolean;
+  stringFormat?: DATE_FORMATS;
+};
+
+/**
+ * Format date into local time and apply the string formatting if given
+ * else the date object is parsed into local time
+ *
+ * @param date date which needs to be parsed or formatted
+ * @param stringFormat string of tokens to format the date, check https://day.js.org/docs/en/display/format#docsNav
+ * @returns string if stringFormat is passed else a date object
+ */
+export const formatDate = (
+  date: string | Date,
+  { asTimestamp, stringFormat, suffix }: formatDateOptions = {}
+) => {
+  const localDate = dayjs(date).local();
+  if (asTimestamp) return localDate.valueOf().toString();
+  const formattedDate = stringFormat
+    ? localDate.format(stringFormat)
+    : localDate;
+  return suffix ? `${formattedDate}${suffix}` : formattedDate;
+};
+
+/**
+ * Parse the given date relative to the user locale
+ *
+ * @param date
+ * @param fromNow give relative date from now
+ * @returns relative date as string
+ */
+export const relativeDateTime = (date: string | Date, fromNow = false) => {
+  const t = dayjs.utc(date).local();
+  const n = dayjs();
+
+  if (n.isSame(t, "day") || fromNow) {
+    return t.fromNow();
+  }
+
+  if (n.diff(t, "days") === 1) {
+    return `${t.format(DATE_FORMATS.TIME)} yesterday`;
+  }
+
+  return t.format(DATE_FORMATS.LONG);
+};
+
+/**
+ *
+ * @param value date to check
+ * @returns boolean check if value is a valid date
+ */
+export const isValidDate = (value: DateConfig) => dayjs(value).isValid();
+
+/**
+ *
+ * @param value date to convert
+ * @returns string of ISO date
+ */
+export const dateToISOString = (value: DateConfig) =>
+  dayjs(value).toISOString();
+
+/**
+ *
+ * @param value date to convert
+ * @returns Native date object
+ */
+export const dateToJSDate = (value: DateConfig) => dayjs(value).toDate();
+
+/**
+ *
+ * @param value date to format
+ * @param format custom string format for the date
+ * @returns formatted string of the date
+ */
+export const customDateFormattor = (value: DateConfig, format: string) =>
```

Comment:
Why is this needed ?

Reply:
@Prashanth-Nelli has used a lot of custom formats for displaying dates in getUpdatedFormat() function of src/components/Canary/CanaryStatusChart/index.tsx. These formats are different from the ones mentioned in https://github.com/flanksource/flanksource-ui/pull/456#discussion_r976382121. 

Hence I added a separate option for displaying these custom formats. Let me know if we there is a different requirement here.

Reply:
@sagarrchauhan we want to standardize on date formats, not have different components using different formats